data_tools/firebase_client_wrapper.py

```diff
@@ -189,6 +189,41 @@ def _delete_collection(coll_ref, batch_size):
     if deleted >= batch_size:
         return _delete_collection(coll_ref, batch_size)
 
+
+def delete_unchecked_messages_in_segment(dataset_id, segment_index=None):
+    if segment_index is not None and segment_index != 1:
+        segment_id = f'{dataset_id}_{segment_index}'
+    else:
+        segment_id = dataset_id
+
+    messages = get_segment_messages(dataset_id, segment_index)
+    messages.sort(key=lambda msg: msg["SequenceNumber"])
+    for msg in messages:
+        # Get the latest label from each scheme
+        latest_labels = dict()  # of scheme id -> label
+        for label in msg["Labels"]:
+            if label["SchemeID"] not in latest_labels:
+                latest_labels[label["SchemeID"]] = label
+
+        manually_coded = False
+        for label in latest_labels.values():
+            if label["Checked"]:
+                manually_coded = True
+
+        if not manually_coded:
+            print(f"Deleting message with sequence number: {msg['SequenceNumber']}")
+            client.document(f"datasets/{segment_id}/messages/{msg['MessageID']}").delete()
```

Comment:
Could we count the number we're deleting to report at the end of the run?

Reply:
added